src/orion/benchmark/task/task_wrapper.py

```diff
@@ -0,0 +1,164 @@
+
+logger = get_logger(__name__)
+
+
+TaskType = TypeVar("TaskType", bound=BaseTask)
+InputType = TypeVar("InputType")
+
+
+class TaskWrapper(BaseTask, Generic[TaskType], ABC):
+    """ ABC for a Wrapper around a Task.
+
+    Wrappers could be used to modify the search space, trials, or the results of the task.
+    """
+
+    def __init__(self, task: TaskType, max_trials: int = None):
+        # TODO: Should Wrappers pass the `task` as an argument to the super() constructor?
+        # When/How are these objects deserialized?
+        if isinstance(task, dict):
+            if len(task) != 1:
+                raise ValueError(
+                    f"Expected task configuration dict to have only one key (the task name), but "
+                    f"got {task} instead."
+                )
+            task_name, task_config = task.popitem()
+            task = BenchmarkTask(of_type=task_name, **task_config)
+        super().__init__(max_trials=max_trials or task.max_trials)
+        self.task: TaskType = task
+
+    def call(self, x: Any) -> List[Dict]:
+        return self.task.call(x)
+
+    def get_search_space(self) -> Dict[str, str]:
+        return self.task.get_search_space()
+
+    @property
+    def unwrapped(self) -> Union[TaskType, BaseTask]:
+        """ Returns the 'unwrapped' task. """
+        return self.task.unwrapped
+
+    @property
+    def configuration(self) -> Dict[str, Any]:
+        """Return the configuration of the task wrapper (including the wrapped task's configuration).
+        """
+        return {
+            type(self).__qualname__: {
+                "task": self.task.configuration,
+                "max_trials": self.max_trials,
+            }
+        }
+
+
+HParamsType = TypeVar("HParamsType")
+
+
+class FixTaskDimensionsWrapper(TaskWrapper[TaskType]):
+    """ Wrapper around a Task that fixes the values of some of its input dimensions.
+    """
+
+    def __init__(self, task: TaskType, fixed_dims: Dict[str, Any], max_trials: int = None):
+        """ Wraps the provided task, fixing the dimensions in `fixed_dims`. """
+        super().__init__(task=task, max_trials=max_trials)
+        self.fixed_dims = fixed_dims
+        # The whole space, from the wrapped task.
+        self._full_space: Space = SpaceBuilder().build(self.task.get_search_space())
+
+        for dimension_name in fixed_dims:
+            if dimension_name not in self._full_space:
+                raise ValueError(
+                    f"Can't fix dimension '{dimension_name}' because it isn't in the task's space: "
+                    f"{self._full_space}."
+                )
+
+        # The part of the space that can be modified.
+        self._space: Space = SpaceBuilder().build(self.get_search_space())
+
+    def call(self, x: Any) -> List[Dict]:
+        """ Calls the wrapped task, adding/changing some dimensions in `x` if necessary. """
+        trial: Trial
+        if isinstance(x, np.ndarray):
```

Comment:
Same comment here about the support for different types of x.

Reply:
Hmm I'm starting to think that this `TaskWrapper` doesn't really have anything to do with the rest of this PR.
This is a wontfix for me atm.

Reply:
Oh, I didn't realize it is not used anywhere in this PR. I would propose you move this class and associated tests to another PR and we can look at it in more details when we have time for it.

Reply:
Fixed: Removed this file and test from this PR.